lib/hanami/helpers/form_helper.rb

```diff
@@ -0,0 +1,213 @@
+      #   @param attributes [Hash] the HTML attributes for the form tag
+      #   @yieldparam [FormHelper::FormBuilder] f the form builder
+      #
+      # @overload form_for(url, values: _form_for_values, params: _form_for_params, **attributes)
+      #   @param url [String] the URL for submitting the form
+      #   @param values [Hash] values to be used for populating form field values; optional,
+      #     defaults to the template's locals or to a part's `{name => self}`
+      #   @param params [Hash] request param values to be used for populating form field values;
+      #     these are used in preference over the `values`; optional, defaults to the current
+      #     request's params
+      #   @param attributes [Hash] the HTML attributes for the form tag
+      #   @yieldparam [FormHelper::FormBuilder] f the form builder
+      #
+      # @return [String] the form HTML
+      #
+      # @see FormHelper
+      # @see FormHelper::FormBuilder
+      #
+      # @example Basic usage
+      #   <%= form_for("book", "/books", class: "form-horizontal") do |f| %>
+      #     <div>
+      #       <%= f.label "title" %>
+      #       <%= f.text_field "title", class: "form-control" %>
+      #     </div>
+      #
+      #     <%= f.submit "Create" %>
+      #   <% end %>
+      #
+      #   =>
+      #   <form action="/books" method="POST" accept-charset="utf-8" class="form-horizontal">
+      #     <input type="hidden" name="_csrf_token" value="920cd5bfaecc6e58368950e790f2f7b4e5561eeeab230aa1b7de1b1f40ea7d5d">
+      #     <div>
+      #       <label for="book-title">Title</label>
+      #       <input type="text" name="book[title]" id="book-title" value="Test Driven Development">
+      #     </div>
+      #
+      #     <button type="submit">Create</button>
+      #   </form>
+      #
+      # @example Without base name
+      #
+      #   <%= form_for("/books", class: "form-horizontal") do |f| %>
+      #     <div>
+      #       <%= f.label "books.title" %>
+      #       <%= f.text_field "books.title", class: "form-control" %>
+      #     </div>
+      #
+      #     <%= f.submit "Create" %>
+      #   <% end %>
+      #
+      #   =>
+      #   <form action="/books" method="POST" accept-charset="utf-8" class="form-horizontal">
+      #     <input type="hidden" name="_csrf_token" value="920cd5bfaecc6e58368950e790f2f7b4e5561eeeab230aa1b7de1b1f40ea7d5d">
+      #     <div>
+      #       <label for="book-title">Title</label>
+      #       <input type="text" name="book[title]" id="book-title" value="Test Driven Development">
+      #     </div>
+      #
+      #     <button type="submit">Create</button>
+      #   </form>
+      #
+      # @example Method override
+      #   <%= form_for("/books/123", method: :put) do |f|
+      #     <%= f.text_field "book.title" %>
+      #     <%= f.submit "Update" %>
+      #   <% end %>
+      #
+      #   =>
+      #   <form action="/books/123" accept-charset="utf-8" id="book-form" method="POST">
```

Comment:
```suggestion
      #   <form action="/books/123" accept-charset="utf-8" method="POST">
```

It can't guess the `id`, without the `base_name` argument.